src/plugins/messenger/index.tsx

```diff
@@ -22,10 +25,10 @@ const getMessengerPayload = (message: IMessage, config: IWebchatConfig) => {
             return _facebook;
         }
 
-        return _webchat;
+        return transformPointers(_webchat,useDefaultReplyCompatibilityMode);
```

Comment:
Now the code will always run through the "hotfix" code which can handle "being disabled" in addition.

I'd prefer to "pick the right object" first and then conditionally apply the `transformPointers` function on top of it. You can make use of IIFEs to get the data object first before deciding whether to return or transform it.


```javascript
() => {
  const theCorrectDataObject = (() => {
    // all the "pick the right object" logic goes here
  })();

  if (compatibilityMode)
    return transform(theCorrectDataObject);

  return theCorrectDataObject;
}
```